src/Hocon.Configuration/Config.cs

```diff
@@ -54,27 +66,12 @@ public Config(HoconRoot root) : base(root?.Value, root?.Substitutions ?? Enumera
         /// <summary>
         ///     The configuration used as a secondary source.
         /// </summary>
-        public Config Fallback { get; private set; }
+        public Config Fallback { get; }
 
         /// <summary>
         ///     The root node of this configuration section
         /// </summary>
-        public virtual HoconValue Root
-        {
-            get
-            {
-                var elements = new List<IHoconElement>();
-                for (var config = this; config != null; config = config.Fallback)
-                {
-                    elements.AddRange(config.Value);
-                }
-
-                var aggregated = new HoconValue(null);
-                aggregated.AddRange(elements.AsEnumerable().Reverse());
-
-                return aggregated;
-            }
-        }
+        public virtual HoconValue Root { get; }
```

Comment:
So this is where we will safe on allocations. `Root` is immutable and calculated only once, in constructor

Reply:
:+1: